### PR TITLE
refactor(api): remove `BOOL` macro

### PR DESCRIPTION
**Pull request by famiu**
Remove redundant `BOOL` macro that does the same thing as `BOOLEAN_OBJ`.

### CODE REVIEW

File: src/nvim/api/buffer.c [hunk 1]
```diff
@@ -886,7 +886,7 @@ Object nvim_buf_get_var(Buffer buffer, String name, Error *err)
   buf_T *buf = find_buffer_by_handle(buffer, err);
 
   if (!buf) {
-    return (Object)OBJECT_INIT;
+    return NIL;
```
Comment (lewis6991) [inline src/nvim/api/buffer.c:889]: This is quite a broad macro name. Could we rename this to `OBJECT_NIL` or `NIL_OBJECT`?

Comment (famiu) [reply src/nvim/api/buffer.c:889]: I am in agreement with this. `NIL` is pretty vague. @bfredl thoughts?

Comment (zeertzjq) [reply src/nvim/api/buffer.c:889]: I prefer keeping `OBJECT_INIT`. It's more consistent with other macros.

Comment (lewis6991) [reply src/nvim/api/buffer.c:889]: `OBJECT_NIL` is more consistent with `OBJECT_INIT`. `NIL_OBJ` would be consistent with `BOOLEAN_OBJ`.

Comment (famiu) [reply src/nvim/api/buffer.c:889]: > I prefer keeping `OBJECT_INIT`. It's more consistent with other macros.

That's also an option. But I would prefer `NIL_OBJ`

Comment (famiu) [reply src/nvim/api/buffer.c:889]: There's also many more instances of `(Object)OBJECT_INIT` (currently `NIL`) than `OBJECT_INIT` by itself, another factor to consider.

Comment (famiu) [reply src/nvim/api/buffer.c:889]: Ah, seems like we need* `OBJECT_INIT` to initialize static `Object`s. We can manually provide `{ .type = kObjectTypeNil }` initializer but `OBJECT_INIT` is clearer. Regardless, I still think `NIL` should be renamed to `NIL_OBJ` if we want to keep it, or remove `NIL` entirely and replace it with `(Object)OBJECT_INIT`

Comment (zeertzjq) [reply src/nvim/api/buffer.c:889]: So just leave `OBJECT_INIT` alone and change `NIL` and `(Object)OBJECT_INIT` to `NIL_OBJ`.

Comment (famiu) [reply src/nvim/api/buffer.c:889]: Alright

Comment (bfredl) [reply src/nvim/api/buffer.c:889]: changing `NIL` to `NIL_OBJ` sounds like like unnecessary churn. It is fine as it is.

Comment (famiu) [reply src/nvim/api/buffer.c:889]: alright then

**Pull request merged**